server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java

```diff
@@ -210,7 +210,9 @@ private Instant parseDateTime(String value, ZoneId timeZone, boolean roundUpIfNo
                 return DateFormatters.from(formatter.parse(value)).toInstant();
             } else {
                 TemporalAccessor accessor = formatter.parse(value);
-                ZoneId zoneId = TemporalQueries.zone().queryFrom(accessor);
```

Comment:
The docs describe the `zone()` query as "A lenient query for the ZoneId, falling back to the ZoneOffset", and how do we feel about leniency?

Reply:
while this would fix the use case where a mapping expects an offset but the problem will re-highlight itself whe na mapping is using zoneId?


```
 public void testParseOffset() {
        // Format string from #76415
        DocValueFormat.DateTime parsesZone = new DocValueFormat.DateTime(
            DateFormatter.forPattern("uuuu-MM-dd'T'HH:mm:ss.SSSSSSSSSVV"),
            ZoneOffset.UTC,
            Resolution.MILLISECONDS
        );
        long expected = 1628719200000L;
        ZonedDateTime sample = ZonedDateTime.of(2021, 8, 12, 0, 0, 0, 0, ZoneId.ofOffset("", ZoneOffset.ofHours(2)));
        assertEquals("GUARD: wrong initial millis", expected, sample.toEpochSecond() * 1000);
        //assertEquals("GUARD: wrong initial string", "2021-08-12T00:00:00.000000000+02:00", parsesZone.format(expected));
        long actualMillis = parsesZone.parseLong(
            "2021-08-12T00:00:00.000000000CET",
            false,
            () -> { throw new UnsupportedOperationException("don't use now"); }
        );
        assertEquals(expected, actualMillis);
    }
```

Reply:
actually I just tested this and it would work..

Reply:
That's a good test though, we should add it to the suite.  Do you mind if I just copy it in?

Reply:
I don't mind it at all :) it is a copy of your test with just  patterns changed :D 
problem is that it is prone to DST changes. `ZoneId.ofOffset("", ZoneOffset.ofHours(2)));` and `2021-08-12T00:00:00.000000000+02:00` will have to account for this